smartrent/badmagic
# ENG: Add HTML Response View

# Description
Found reading the raw html particularly frustrating when creating new routes and you have error responses from Phoenix which are unreadable. I was copying the text to a file to view in the browser but this is much faster 😄  

This adds an option to view a rendered html response for a route. It also adds a basic local test server for development.

The response is not rendered by default and it is rendered in an iframe for sandboxing.

# Steps to test

1. Follow steps in README.md to setup the local `example` folder.
2. Go to [http://localhost:3005](http://localhost:3005).
3. Try the `/views/breeds.html` route.
4. Observe raw response.
5. Click "Render HTML" button.
6. Observe rendered response.

# Screenshots

![html-preview](https://user-images.githubusercontent.com/11529600/142028094-64b2b32b-b660-4194-aa16-d0b9632cac76.gif)
